docs/gallery/general/plot_resources.py

```diff
@@ -0,0 +1,396 @@
+# (i.e., :py:class:`~pynwb.resources.Object.relative_path` must be the shortest possible path and
+# as such cannot contain any objects with a ``data_type`` and associated ``object_id``).
+#
+# A common example would be with the :py:class:`~pynwb.table.DynamicTable` class, which holds
+# :py:class:`~pynwb.table.VectorData` objects as columns. If we wanted to add an external
+# reference on a column from a :py:class:`~pynwb.table.DynamicTable`, then we would use the
+# column as the object and not the :py:class:`~pynwb.table.DynamicTable` (Refer to rule 9).
+
+genotypes = DynamicTable(name='genotypes', description='My genotypes')
+genotypes.add_column(name='genotype_name', description="Name of genotypes")
+genotypes.add_row(id=0, genotype_name='Rorb')
+er.add_ref(
+    file=file,
+    container=genotypes,
+    attribute='genotype_name',
+    key='Rorb',
+    entity_id='MGI:1346434',
+    entity_uri='http://www.informatics.jax.org/marker/MGI:1343464'
+)
+
+# Note: :py:func:`~pynwb.resources.HERD.add_ref` internally resolves the object
+# to the closest parent, so that ``er.add_ref(container=genotypes, attribute='genotype_name')`` and
+# ``er.add_ref(container=genotypes.genotype_name, attribute=None)`` will ultimately both use the ``object_id``
+# of the ``genotypes.genotype_name`` :py:class:`~pynwb.table.VectorData` column and
+# not the object_id of the genotypes table.
+
+###############################################################################
+# Using the add_ref method without the file parameter.
+# ------------------------------------------------------
+# Even though :py:class:`~pynwb.resources.File` is required to create/add a new reference,
+# the user can omit the file parameter if the :py:class:`~pynwb.resources.Object` has a file
+# in its parent hierarchy.
+
+col1 = VectorData(
+    name='Species_Data',
+    description='species from NCBI and Ensemble',
+    data=['Homo sapiens', 'Ursus arctos horribilis'],
+)
+
+# Create a DynamicTable with this column and set the table parent to the file object created earlier
+species = DynamicTable(name='species', description='My species', columns=[col1])
+species.parent = file
+
+er.add_ref(
+    container=species,
+    attribute='Species_Data',
+    key='Ursus arctos horribilis',
+    entity_id='NCBI_TAXON:116960',
+    entity_uri='https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&id'
+)
+
+###############################################################################
+# Visualize HERD
+# ------------------------------------------------------
+# Users can visualize `~pynwb.resources.HERD` as a flattened table or
+# as separate tables.
+
+# `~pynwb.resources.HERD` as a flattened table
+er.to_dataframe()
+
+# The individual interlinked tables:
+er.files.to_dataframe()
+er.objects.to_dataframe()
+er.entities.to_dataframe()
+er.keys.to_dataframe()
+er.object_keys.to_dataframe()
+er.entity_keys.to_dataframe()
```

Comment:
This will only show the last table (here entity_keys) in the output. You will either need to call `display(er.files.to_dataframe())` for each of the DataFrames or separate them into cell.